openfisca_core/simulations/simulation.py

```diff
@@ -412,17 +416,12 @@ def invalidate_cache_entry(self, variable: str, period):
         self.invalidated_caches.add(Cache(variable, period))
 
     def invalidate_spiral_variables(self, variable: str):
-        # Visit the stack, from the bottom (most recent) up; we know that we'll find
-        # the variable implicated in the spiral (max_spiral_loops+1) times; we keep the
-        # intermediate values computed (to avoid impacting performance) but we mark them
-        # for deletion from the cache once the calculation ends.
-        count = 0
-        for frame in reversed(self.tracer.stack):
-            self.invalidate_cache_entry(str(frame["name"]), frame["period"])
-            if frame["name"] == variable:
-                count += 1
-                if count > self.max_spiral_loops:
-                    break
+        invalidate_entries = False
```

Comment:
I notice that this delta removes a comment. The comment noted that the stack was visited in reverse order; I'd assume that was a significant decision. I'm a bit worried to see this delta use a different order AND remove the comment.

"The tests still pass, so it must be OK" is an objection that I could understand, but perhaps in the context of OF-Core it's not as convincing as it would be in other contexts…

Reply:
Thanks for your comment @Morendil. Indeed, I changed the logic because the `break` logic was not general enough.
In order to avoid silent regressions, I updated and improved the relevant tests in #1206.

Reply:
I realized that the intention could be improved incrementally. I update the function code. In my opinion, it is a much clearer version.